### PR TITLE
NOD - Add "Create NOD" Method to DecisionReview::Service

### PR DESCRIPTION
**Pull request by edmangimelli**
(for the [new NOD product](https://github.com/department-of-veterans-affairs/va.gov-team/issues/2140))

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -82,6 +82,11 @@ def middle_name
     identity.middle_name || (mhv_icn.present? ? mpi&.profile&.given_names.to_a[1..-1]&.join(' ').presence : nil)
   end
 
+  # returns a string or nil (never returns a ".blank?" string)
+  def middle_initial(append = '')
```
Comment (bosawt) [inline app/models/user.rb:86]: Curious what the need of `append` functionality is here? It doesn't look like it's used

Comment (bosawt) [reply app/models/user.rb:86]: Overall we're planning on moving the presentation layer outside of the User model (for example, providing just the `middle_name` and leaving it up to classes outside of `User` to extract the Middle Name from it), so my preference would be to just have the line in `lib/decision_review/service.rb `: `'X-VA-Middle-Initial' => middle_name.to_s.strip.first.upcase`.

However, especially if you foresee other places needing `middle_initial` in the future, I won't push back too hard if you prefer it here

**Review by bosawt — COMMENTED** [review]
Just a quick couple of comments about the change in the `User` model!

File: app/models/user.rb (hunk 1 above)
Comment (edmangimelli) [reply app/models/user.rb:86]: at one point I was thinking I'd need append, but, you're right, I didn't end up using it. I do think it's a nice feature, though: (only append if there is a middle initial)

```
user.middle_name # Jacob
"#{user.first_name}--#{user.middle_initial('--'}#{user.last_name}" # "John--J--Jingleheimer"

user.middle_name # nil
"#{user.first_name}--#{user.middle_initial('--'}#{user.last_name}" # "John--Jingleheimer"
```

but, not a hill i want to die on 👍

Comment (bosawt) [reply app/models/user.rb]: Oh I see, ya that totally makes sense. If we do feel like we need that functionality and/or start needing Middle Initial in multiple places, I think we can revisit whether it belongs in the User model for sure.

**Review by bosawt — APPROVED** [review]
Considering there aren't changes left in Identity-related code anymore, that's an easy approval on my end!

Also while I was reviewing I just looked through the usage of the different `User` fields, and it looks like the function chains won't raise no-method errors regardless of presence of these fields, so passes my quick sanity check

**Review by thilton-oddball — APPROVED** [review]
Looks good!

**Pull request merged**